**Table S1. Summary of Riptide and conventional Illumina libraries used in this analysis**    
**Table S2. Patterns of insertions and deletions in Riptide and non-Riptide datasets**  
**Table S3. Expected sequencing depth of Riptide and conventional Illumina libraries**  
**Table S4. Summary of Riptide vs. non-Riptide SPAdes assemblies**  
**Table S5. Aspergillus SPAdes assembly statistics**  
**Table S6. Brugia SPAdes assembly statistics**  
**Table S7. E. coli SPAdes assembly statistics**  
**Table S8. Klebsiella SPAdes assembly statistics**   
**Table S9. Plasmid pB171 SPAdes assembly statistics**  
**Table S10. Plasmodium SPAdes assembly statistics**  
**Table S11. Theileria SPAdes assembly statistics**  
**Table S12. Gap positions in de novo E. coli genome assemblies**  
**Table S13. Classification of metagenome datasets using kraken2**  
